src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -56,33 +69,46 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
-        public virtual HostSecrets GetHostSecrets()
+        public virtual HostSecretsInfo GetHostSecrets()
         {
             if (_hostSecrets == null)
             {
                 string secretFilePath = Path.Combine(_secretsPath, ScriptConstants.HostMetadataFileName);
+                var hostSecrets = new HostSecrets();
+
                 if (File.Exists(secretFilePath))
                 {
                     // load the secrets file
                     string secretsJson = File.ReadAllText(secretFilePath);
-                    _hostSecrets = JsonConvert.DeserializeObject<HostSecrets>(secretsJson);
+                    hostSecrets = ScriptSecretSerializer.DeserializeHostSecrets(secretsJson);
                 }
                 else
                 {
                     // initialize with new secrets and save it
-                    _hostSecrets = new HostSecrets
+                    hostSecrets = new HostSecrets
```

Comment:
Looked at removing this, but keeping it in place allows for a clear API for the serializers (where they're work with `HostSecrets`, which contains a list of `Key` objects) as well as the secret consumers (where they work with `HostSecretsInfo`, which is a simpler type that makes more sense to consumers). Also trying to avoid having the `Key` returned to consumers, since if we did, we'd likely push the requirement to have a simpler type (e.g. `KeyInfo`) for external consumers.